### PR TITLE
Fixed breadcrumbs calculation and actions flow

### PR DESCRIPTION
**Pull request by skjnldsv**
Needs https://github.com/nextcloud/gallery/pull/336

dev skjnldsv com_40402_apps_files__di156r _
dev skjnldsv com_40402_apps_files__dir _

### CODE REVIEW

**Comment by MorrisJobke** [comment]
In the files app the gallery button is not pushed to the left anymore once you open the sidebar. In the past the button was moved to the left of the sidebar once it was opened. This may be crucial in the future once the sidebar is open per default for wide screens.

File: apps/files/css/files.scss [hunk 1]
```diff
@@ -21,6 +21,11 @@
 .actions.creatable {
 	position: relative;
```
Comment (pixelipo) [inline apps/files/css/files.scss]: I guess we don't need this anymore.

File: apps/files/css/files.scss [hunk 2]
```diff
@@ -21,6 +21,11 @@
...
 	z-index: -30;
+	display: flex;
+	flex: 1 1;
+	.button:not(:last-child) {
+		margin-right: 3px;
```
Comment (pixelipo) [inline apps/files/css/files.scss]: any reason for this? AFAICS, it's only applies to `button.new`, which doesn't need it (`#uploadprogresswrapper` already has a left margin).

**Review by pixelipo — APPROVED** [review]
Exactly how it should be solved (few CSS nitpicks aside)

Some work is needed in the gallery app PR.

File: apps/files/css/files.scss (hunk 2 above)
Comment (skjnldsv) [reply apps/files/css/files.scss]: Yes, in case of other apps want to add their buttons, let them do it without needing to define specific margins :)

Comment (pixelipo) [reply apps/files/css/files.scss]: OK, but then you can remove the margin from `#uploadprogresswrapper`

Comment (pixelipo) [reply apps/files/css/files.scss]: also, it should be `#controls .button`, not `.actions.createable`

Comment (skjnldsv) [reply apps/files/css/files.scss]: All buttons should be in the .actions div now. Far better thann having buttons everywhere :)

Comment (pixelipo) [reply apps/files/css/files.scss]: perhaps, but definitely not in `.createable` ;)

**Comment by pixelipo** [comment]
Should @MorrisJobke 's issue be fixed in this PR as well? I'm made a PR to this branch with a proposed solution to it: #7484

**Review by MorrisJobke — APPROVED** [review]
Together with #7484 this works fine 👍

**Comment by skjnldsv** [comment]
Okay, fixed zindex, rebased and seems fine! :)

**Comment by skjnldsv** [comment]
Test fixed! There was an issue from #7051.... The test was basically useless ^^

**Pull request merged**